woocommerce/woocommerce-gutenberg-products-block
# Add Product Collection Full Grid pattern

## What

Add Product Collection Full Grid pattern.

Additionally, when implementing I realized the Product Button doesn't react on the "Text align" property. Be mindful in this context the initial purpose was to align the whole button (as per https://github.com/woocommerce/woocommerce-blocks/pull/7816) which was preserved in this PR.

Fixes https://github.com/woocommerce/woocommerce-blocks/issues/9369

## Why

This is one of the patterns from a set of patterns we'd like to provide for Product Collection

## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

0. At this moment Product Collection is still experimental feature (PR to make it Core feature: https://github.com/woocommerce/woocommerce-blocks/pull/10524), hence the pattern is not included in the inserter. In order to test the pattern please remove this line: https://github.com/woocommerce/woocommerce-blocks/pull/10660/files#diff-d26dedf0a7efa88eae30cbe732192f2769376839845875e6e57505d33f19653cR6.
1. Go to Editor
3. Insert "Product Collection Full Grid" pattern
4. Save and go to frontend
5. Verify it looks like in the screenshots below

Known limitations:
- Product Price cannot be styled according to designs. This work is being covered in https://github.com/woocommerce/woocommerce-blocks/pull/9398


## Screenshots or screencast



Reference | Editor | Frontend |
| ------ | ----- | ---|
| ![image](https://github.com/woocommerce/woocommerce-blocks/assets/20098064/b8efebdb-dbfc-4181-93bf-cc0bcd8c144c) |    <img width="802" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/20098064/f3d37fe3-e1f0-4a85-8894-2bcd98f6d46f">   |    <img width="702" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/20098064/3a3f35c7-9ef0-4eab-96b7-1bb834ae7ad6">   |

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> Product Collection: new pattern "Product Collection Full Grid"
